ush/get_obs.py

```diff
@@ -0,0 +1,1004 @@
+                        elif obtype == 'NOHRSC':
+                            yyyymmddhh_str = dt.datetime.strftime(yyyymmddhh, '%Y%m%d%H')
+                            fn_raw = 'sfav2_CONUS_' + accum_obs_formatted + 'h_' + yyyymmddhh_str + '_grid184.grb2'
+                        elif obtype == 'MRMS':
+                            hr = yyyymmddhh.hour
+                            fn_raw = fields_in_filenames[i] + '_' + levels_in_filenames[i] \
+                                   + '_' + yyyymmdd_task_str + '-' + f'{hr:02d}' + '0000.grib2'
+                            fn_raw = os.path.join('topofhour', fn_raw)
+                        elif obtype == 'NDAS':
+                            time_ago = yyyymmddhh_arcv - yyyymmddhh
+                            hrs_ago = int(time_ago.seconds/3600)
+                            hh_arcv_str = dt.datetime.strftime(yyyymmddhh_arcv, '%H')
+                            fn_raw = 'nam.t' + hh_arcv_str + 'z.prepbufr.tm' + f'{hrs_ago:02d}' + '.nr'
+                        fp_raw = os.path.join(arcv_dir_raw, fn_raw)
+
+                        # Get the full path to the final processed obs file (fp_proc) we want to
+                        # create.
+                        indx = obs_retrieve_times_crnt_day.index(yyyymmddhh)
+                        fp_proc = all_fp_proc_dict[fg][indx]
+
+                        # Make sure the directory in which the processed file will be created exists.
+                        dir_proc = os.path.dirname(fp_proc)
+                        Path(dir_proc).mkdir(parents=True, exist_ok=True)
+
+                        msg = dedent(f"""
+                            Creating the processed obs file
+                                {fp_proc}
+                            from the raw file
+                                {fp_raw}
+                            ...
+                            """)
+                        logging.info(msg)
+
+                        yyyymmdd = yyyymmddhh.replace(hour=0, minute=0, second=0)
+                        # CCPA files for 1-hour accumulation have incorrect metadata in the files
+                        # under the "00" directory from 20180718 to 20210504.  After the data is
+                        # pulled, reorganize into correct yyyymmdd structure.
+                        if (obtype == 'CCPA') and \
+                           ((yyyymmdd >= yyyymmdd_bad_metadata_start) and (yyyymmdd <= yyyymmdd_bad_metadata_end)) and \
+                           (((hr >= 19) and (hr <= 23)) or (hr == 0)):
+                            cmd = ' '.join(['wgrib2', fp_raw, '-set_date -24hr -grib', fp_proc, '-s'])
+                            result = subprocess.run(cmd, shell=True, capture_output=True, text=True)
+                        elif mv_or_cp == 'mv':
+                            shutil.move(fp_raw, fp_proc)
+                        elif mv_or_cp == 'cp':
+                            shutil.copy(fp_raw, fp_proc)
```

Comment:
The above logic creating this `mv_or_cp` variable was unnecessary, we can do the same here in way fewer lines.

Also, I'm pretty sure this is supposed to be an `if` not an `elif` for the move logic, otherwise it will never be applied for CCPA, no? Unless there's something about CCPA specifically I missed.

```suggestion

                        if vx_config['REMOVE_RAW_OBS_' + obtype]:
                            shutil.move(fp_raw, fp_proc)
                        else:
                            shutil.copy(fp_raw, fp_proc)
```

Reply:
It should be an `elif` since the `if` is entered only if the obs type is CCPA **and** the date falls between those two limits.  So if the date isn't between the two limits, the `elif` or the `else` will execute for CCPA obs.

If it's changed into an `if`, then for CCPA and a date that is between the limits, the wgrib2 command will first create a corrected processed file, but that will then get overwritten by the bad raw file.